posthog/hogql_queries/utils/timestamp_utils.py

```diff
@@ -135,24 +135,32 @@ def _get_earliest_timestamp_from_node(team: Team, node: Union[EventsNode, Action
 
 
 def get_earliest_timestamp_from_series(
-    team: Team, series: list[Union[EventsNode, ActionsNode, DataWarehouseNode]]
+    team: Team, series: list[Union[EventsNode, ActionsNode, DataWarehouseNode, GroupNode]]
 ) -> datetime:
     """
     Get the earliest timestamp for specific events/actions in a series.
     This is used for "All time" date filtering - each event/action is queried
     for its own earliest timestamp, and the minimum is returned.
 
     :param team: The team
-    :param series: A list of series nodes (EventsNode, ActionsNode, or DataWarehouseNode)
+    :param series: A list of series nodes (EventsNode, ActionsNode, DataWarehouseNode, or GroupNode)
     :return: The earliest timestamp across all series
     """
+    # Expand GroupNode nodes into individual nodes
+    nodes: list[Union[EventsNode, ActionsNode, DataWarehouseNode]] = []
+    for node in series:
```

Comment:
side-note: it should be more performant to get the earliest timestamp in a single query. wondering why we have multiple queries here in the first place. maybe i'm missing something.

Reply:
Yup definitely out of scope for this one. And yeah the caching thing makes sense.